### ISSUE TITLE
Provide ability to access console / tty of podman machine via socket

### ISSUE THREAD

**Issue by mairin** [containers/podman#20466]
### Feature request description

Sometimes if you update your podman machine or otherwise break it, you might render it in a state where it doesn't boot such that you can access a shell. You might not understand why it's broken or if it's recoverable. You could create a new podman machine but you might lose what you had set up in the old one.

### Suggest potential solution

For these situations, it would be helpful to be able to access the TTYs for the VM to try to get some output and see what is going on. This would be regardless is the VM backend is qemu, hyper-v, etc.

We would like to provide this via Podman Desktop but need access to the TTYs via socket.

### Additional context

Correponding RFE for Podman Desktop is here:
https://github.com/containers/podman-desktop/issues/4493

**Comment by rhatdan** [comment]
@baude @ashley-cui @gbraad @n1hility PTAL

**Comment by ashley-cui** [comment]
Something like showing the tty output for `podman machine start` in the terminal that we're currently running it from? I remember qemu has a flag for this, but we'd have to find the equiv on the other providers.

**Comment by mairin** [comment]
@ashley-cui if that's preboot output then yeh!

**Comment by rhatdan** [comment]
Yes add a `podman machine start --tty `flag and then setup the terminal for the virtualization tool to show the terminal output, if possible.

**Comment by n1hility** [comment]
Just a note that today on the qemu based runtimes right now if you start with --log-level debug it does open a emu graphical window into the console. However unless you set a password (or boot it into single user) you won't be able to login.

**Comment by baude** [comment]
i think with log-level=debug you get a ton of information.  that said,  i am sitting on a PR that @cevich is testing where i have added some logic to ensure gvproxy is running before proceeding.  The same logic could added to vfkit (i believe qemu already does it).  this would separate machine issues from booting issues.

while boot issues are rare, they are usually visible in the debug console.

**Comment by cevich** [comment]
IIRC, you said the 'debug console" is a GUI only thing, eh?

**Comment by rhatdan** [comment]
@baude @cevich Any update?
